dataset_analysis/generate_dataframe.py

```diff
@@ -0,0 +1,462 @@
+                lesion_length_in_level += length_of_lesion
+
+        #compute the average vertical length and volume
+        avg_lesion_length_in_level = 0
+        avg_lesion_volume_in_level = 0
+        if nb_lesions_in_level != 0:
+            avg_lesion_length_in_level = lesion_length_in_level/nb_lesions_in_level
+            avg_lesion_volume_in_level = total_lesion_volume_in_level/nb_lesions_in_level
+
+        #we add this information to the patient_data dictionary
+        patient_data[f"nb_lesions_between_{int(levels[i]):02d}_and_{int(levels[i+1]):02d}"] = nb_lesions_in_level
+        patient_data[f"total_lesion_volume_between_{int(levels[i]):02d}_and_{int(levels[i+1]):02d}"] = total_lesion_volume_in_level
+        patient_data[f"avg_lesion_volume_between_{int(levels[i]):02d}_and_{int(levels[i+1]):02d}"] = avg_lesion_volume_in_level
+        patient_data[f"avg_lesion_length_between_{int(levels[i]):02d}_and_{int(levels[i+1]):02d}"] = avg_lesion_length_in_level
+
+        #we add the lesions which are above the 1st level
+        if int(levels[i]) == 1:
+            #we only have a lower bound
+            lower_bound = np.where(disc_seg_data == levels[i])
+            lower_bound = int(lower_bound[1])*disc_voxel_size_y
+
+            #initialize the lesion_info_per_levels dictionary
+            nb_lesions_above_1 = 0
+            total_lesion_volume_above_1 = 0
+            lesion_length_above_1 = 0
+
+            #now we look for the lesions in this level
+            ## get the lesions
+            lesions = np.unique(lesion_labelled_seg_data)
+            lesions = lesions[lesions != 0]
+            ## iterate over the lesions
+            for lesion in lesions:
+                lesion_voxel = np.where(lesion_labelled_seg_data == lesion)
+                center_of_lesion = np.mean(lesion_voxel, axis=1)
+                volume_of_lesion = len(lesion_voxel[0])*voxel_size_x*voxel_size_y*voxel_size_z
+                top_of_lesion = np.max(lesion_voxel[1])
+                bottom_of_lesion = np.min(lesion_voxel[1])
+                length_of_lesion = (top_of_lesion - bottom_of_lesion)*voxel_size_y
+                #if lesion center is between upper and lower bound, then it is in the level
+                if center_of_lesion[1]*voxel_size_y >= lower_bound:
+                    nb_lesions_above_1 += 1
+                    total_lesion_volume_above_1 += volume_of_lesion
+                    lesion_length_above_1 += length_of_lesion
+
+            #compute the average vertical length and volume
+            avg_lesion_length_above_1 = 0
+            avg_lesion_volume_above_1 = 0
+            if nb_lesions_above_1 != 0:
+                avg_lesion_length_above_1 = lesion_length_above_1/nb_lesions_above_1
+                avg_lesion_volume_above_1 = total_lesion_volume_above_1/nb_lesions_above_1
+
+            # add info to dictionary
+            patient_data[f"nb_lesions_above_01"] = nb_lesions_above_1
```

Comment:
```suggestion
            patient_data[f"nb_lesions_above_C1"] = nb_lesions_above_1
```

Maybe "above_C1"?

Reply:
If I change this, I would need to change all the others as well. Keeping it like this for now.